hack/build-images.sh

```diff
@@ -1,14 +1,75 @@
 #!/bin/bash
 
-set -ex
+set -o errexit
+set -o nounset
+set -o pipefail
 
-prefix=${PREFIX:-${1:-openshift/origin-}}
-version=${VERSION:-${2:-latest}}
-for component in fluentd elasticsearch kibana deployment curator ; do
-    docker build -t "${prefix}logging-${component}:${version}"       ../$component/
+STARTTIME=$(date +%s)
+
+prefix="${PREFIX:-openshift/origin-}"
+version="${VERSION:-latest}"
+os_tag="${OS_PUSH_TAG:-latest}"
+
+source_root=$(dirname "${0}")/..
+
+# update the FROM statement in the Dockerfile to be a specific
+# tag.  Assumes there is only 1 FROM but will change them all
+# if there is more than one
+function create_dockerfile_using_base_tag {
```

Comment:
@smarterclayton what's the logic behind not having the deployer code in this repo? It's the first I've heard of it. Most of this repo is deployer code.